src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -704,17 +741,19 @@ public boolean equals(Object o) {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof VectorArg that)) {
         return false;
       }
-      VectorArg vectorArg = (VectorArg) o;
-      return features == vectorArg.features
-          && stringificationType.equals(vectorArg.stringificationType);
+      return features == that.features
+          && stringificationType.equals(that.stringificationType)
+          // There is only a single instance of StarlarkSemantics per build (see #internerFor).
+          && starlarkSemantics == that.starlarkSemantics;
```

Comment:
@justinhorvitz I adopted your commit, but also added this change on top since using reference equality for `StarlarkSemantics` is cheaper and I find it more consistent with the interning logic above. Let me know if I have missed anything, happy to change this back.

Reply:
It should be fine but if there's no benefit (because we hit the reference equality check in `StarlarkSemantics#equals` anyway) then probably not worth the extra nuance. I'll run another benchmark because I wanted to see if we can get away with a simple weak interner when semantics is present, instead of the mutable active one.

I'll approve this and make changes on import.

Reply:
Also note `StarlarkSemantics` caches its hashcode.

Reply:
The dual strong/weak interner idea works well: 
![image](https://github.com/user-attachments/assets/994bd733-7a80-4fd0-97ad-339a2456e38c)

I'll import that version since it is simpler and also more extensible - if you want to pursue storing the `map_each` function in the `VectorArg`, it works out of the box. Thanks for bearing with me.